"No comment syntax is defined.  Use: " in minibuffer when compiling after test/error

(Thanks, as ever, for CIDER. It's the best and you're the best)

## Expected behavior
File compiles fine, no buffers move or close, minibuffer is not focused, workflow uninterrupted.

## Actual behavior
`*cider-error*` buffer is buried, minibuffer is focused and says `No comment syntax is defined.  Use: `, file may or may not be compiled.

## Steps to reproduce the problem
1) Run a test (`cider-test-run-test`) that has an error (throws)
    - `*cider-test-report*` buffer appears (as expected)
2) Select the `error:` link at the bottom of the buffer
    - `*cider-error*` buffer appears w/ stack-trace in it (as expected)
3) Select the line in your code that caused the error
    - File with errant code it in appears (as expected)
4) Make a change to the code - attempting to fix the error
5) `C-x C-s` `C-c C-k` to save the file and compile the new (hopefully fixed) code
    - :boom: `cider-error` buffer weirdly is buried, minibuffer is focused and says `No comment syntax is defined.  Use: ` ...

## Environment & Version information
GNU Emacs 24.5.1 on Linux gibson 4.8.0-59-generic #64-Ubuntu SMP Thu Jun 29 19:38:34 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

### CIDER version information

```
;; CIDER 0.15.1 (London), nREPL 0.2.12
;; Clojure 1.8.0, Java 1.8.0_111
```

### Lein/Boot version

```
$ lein --version
Leiningen 2.7.2-SNAPSHOT on Java 1.8.0_111 Java HotSpot(TM) 64-Bit Server VM
```

(2) means click on the link that opens the `*cider-error*` buffer, same as (2) in the original reproduction steps

But I apologize, (1) should have said "run a test" like the original, not just load the a buffer.  I don't think it happens unless the `*cider-error*` buffer is buried when the `*cider-test-report*` buffer is also open.

1. make a test like:

```
(deftest assert-test
  (assert false))
```
and `cider-test-run-test`

2. open the `*cider-error*` buffer by clicking the `java.lang.AssertionError: Assert failed: false` link
3. `cider-load-buffer` from the buffer with the test
4. `*cider-error*` buffer is buried, see `No comment syntax is defined. Use:` in minibuffer

I'm still encountering this on 0.18.0

Again steps to reproduce:
1. run a test that causes an error, like add an `(assert false)`
2. click on the link in `*cider-test-report*`: `error: java.lang.AssertionError: Assert failed: false`
3. `cider-load-buffer` a buffer

A small repro file is:
```clj
(ns cider-bug
  (:require [clojure.test :refer :all]))

(deftest test-for-bug
  (is (assert false)))
```

Load/test that buffer, click this link that I have highlighted:
<img src="https://user-images.githubusercontent.com/1058897/50044169-ab6c6400-004d-11e9-8abe-de81caf2ab4a.png" width="400" />
This window appears:
<img src="https://user-images.githubusercontent.com/1058897/50044173-b58e6280-004d-11e9-8b71-327dee28047f.png" width="400" />

Don't close the test or error window.  Go back to the `cider-bug` ns and `cider-load-buffer`.  The `*cider-error*` buffer disappears, but `No comment syntax is defined. Use:` appears in the minibuffer

This happens when my `~/.emacs` file is completely minimal.

The workaround to comment out this line still works: https://github.com/clojure-emacs/cider/blob/3a520ec76cdd5509bf231e49809715f20a87d74b/cider-eval.el#L1060

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@
 * Fix jack-in from inside of remote buffers.
 * [#2454](https://github.com/clojure-emacs/cider/pull/2454): Fix erratic inspector behavior when multiple REPLs are connected
 * [#2467](https://github.com/clojure-emacs/cider/pull/2467): Make generic CIDER ops use any available nREPL connection.
+* [#2105](https://github.com/clojure-emacs/cider/issues/2105): Fix no comment syntax defined message when loading buffer after running a failing test.
 
 ### Changes
 

--- a/cider-eval.el
+++ b/cider-eval.el
@@ -168,7 +168,8 @@ When invoked with a prefix ARG the command doesn't prompt for confirmation."
 (defun cider--quit-error-window ()
   "Buries the `cider-error-buffer' and quits its containing window."
   (when-let* ((error-win (get-buffer-window cider-error-buffer)))
-    (quit-window nil error-win)))
+    (save-excursion
+      (quit-window nil error-win))))
 
 
 ;;; Dealing with compilation (evaluation) errors and warnings
@@ -1055,10 +1056,9 @@ for the project, it is evaluated in both REPLs."
                    (or (eq cider-save-file-on-load t)
                        (y-or-n-p (format "Save file %s? " buffer-file-name))))
           (save-buffer))
-        (save-excursion
-          (remove-overlays nil nil 'cider-temporary t)
-          (cider--clear-compilation-highlights)
-          (cider--quit-error-window))
+        (remove-overlays nil nil 'cider-temporary t)
+        (cider--clear-compilation-highlights)
+        (cider--quit-error-window)
         (let ((filename (buffer-file-name buffer))
               (ns-form  (cider-ns-form)))
           (cider-map-repls :auto

--- a/cider-eval.el
+++ b/cider-eval.el
@@ -1055,9 +1055,10 @@ for the project, it is evaluated in both REPLs."
                    (or (eq cider-save-file-on-load t)
                        (y-or-n-p (format "Save file %s? " buffer-file-name))))
           (save-buffer))
-        (remove-overlays nil nil 'cider-temporary t)
-        (cider--clear-compilation-highlights)
-        (cider--quit-error-window)
+        (save-excursion
+          (remove-overlays nil nil 'cider-temporary t)
+          (cider--clear-compilation-highlights)
+          (cider--quit-error-window))
         (let ((filename (buffer-file-name buffer))
               (ns-form  (cider-ns-form)))
           (cider-map-repls :auto

